kubernetes/cache/cache_handlers.go

```diff
@@ -1,149 +1,50 @@
 package cache
 
 import (
-	"reflect"
-	"strings"
-
-	networking_v1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
-	networking_v1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
-	security_v1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
-	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/tools/cache"
 
-	"github.com/kiali/kiali/kubernetes"
 	"github.com/kiali/kiali/log"
 )
 
+// TODO: Write some units for this.
+
 type RegistryRefreshHandler struct {
 	Cache     KialiCache
-	Namespace string
 	cache.ResourceEventHandler
 }
 
-func NewRegistryHandler(cache KialiCache, namespace string) RegistryRefreshHandler {
-	log.Infof("Adding a RegistryRefreshHandler for [%s]", namespace)
-	return RegistryRefreshHandler{Cache: cache, Namespace: namespace}
+func NewRegistryHandler(cache KialiCache) RegistryRefreshHandler {
+	log.Infof("Adding a RegistryRefreshHandler")
+	return RegistryRefreshHandler{Cache: cache}
 }
 
 func (sh RegistryRefreshHandler) OnAdd(obj interface{}) {
 	sh.Cache.RefreshRegistryStatus()
 }
 
 func (sh RegistryRefreshHandler) OnUpdate(oldObj, newObj interface{}) {
-	resourceVersion1, resourceVersion2 := sh.parseResourceVersion(oldObj, newObj)
-	if resourceVersion1 != resourceVersion2 {
+	var (
+		oldMeta v1.Object
+		newMeta v1.Object
+		err error
+	)
+	
+	if oldMeta, err = meta.Accessor(oldObj); err != nil {
```

Comment:
Oh, nice one, but I remember that there was some "kinda" "bug/feature" in the API machinery, and some fields in the returned object are not populated.

But it makes sense; just a note to test this with detail to avoid side effects.

Reply:
This shouldn't be affected by that issue since it's just using `resourceVersion` and not any of the type meta but +1 on testing to avoid side effects.